eng/AzSdk.reference.targets

```diff
@@ -1,17 +0,0 @@
-<Project ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
-  <ItemGroup>
```

Comment:
Moving these might break some of the data plane projects as we were getting pulled in some cases at https://github.com/Azure/azure-sdk-for-net/blob/master/eng/Directory.Build.Data.targets#L34.

Reply:
just now discussed this with @chidozieononiwu and I told him that I was getting all sorts of error when I first pulled latest upstream and tried working with that file.
But I told him, if that file works fine, I am ok with it, but if it does not, let's not move too many pieces and keep the file in it's original location and squeeze out all the bugs.
Basically I am open to changing the location of this file as long as it works.